### DISCUSSION TITLE
Insert Query Hung in QueryProfiler

### DISCUSSION THREAD

**Discussion by larryluogit** [ClickHouse/ClickHouse#72351] [Q&A]
Insert Queries Running Very Slowly on One Node in a 12-Node Cluster

I am running a 12-node cluster in AWS using ClickHouse 23.8.

```
SELECT version();

┌─version()─┐
│ 23.8.5.16 │
└───────────┘

```

On one particular node, Insert queries (inserting into local replicate table) seem to hang. They remain stuck until they reach the 100-second timeout limit, at which point they are killed.

By querying system.stack_trace, I can see all the Insert queries are stuck in the Query Profiler:

```
sym: StackTrace::StackTrace(ucontext_t const&)
    /opt/clickhouse/bin/clickhouse
DB::(anonymous namespace)::signalHandler(int, siginfo_t*, void*)
    /opt/clickhouse/bin/clickhouse

```
I attempted to disable the Query Profiler by setting the following configuration:

```
<query_profiler_real_time_period_ns>0</query_profiler_real_time_period_ns>
<query_profiler_cpu_time_period_ns>0</query_profiler_cpu_time_period_ns>

```
However, it did not make any difference. Based on [the code](https://github.com/ClickHouse/ClickHouse/blob/9237ab47de40e27e11e25b5b46f036d582ebe2e7/src/Common/QueryProfiler.cpp#L142), it appears that the Query Profiler cannot be completely turned off.

The strange part is that this issue only occurs on one node, while the other 11 nodes are functioning normally.

Any suggestions or insights into what might be causing this would be greatly appreciated!

**Comment by viktordebulat** [comment]
Hey @larryluogit , have you managed to find out root cause? This might be something similar to what I've faced on 25.3
